### PR TITLE
Update references to 1.14 to 2.0

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #91769

There won't be a 1.14 release, so these should be updated.

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

**Review by svekars — APPROVED** [review]

**Comment by lezcano** [comment]
Could you potentially run a (carefult) global replace doing 1.14 -> 2.0?

File: torch/functional.py [hunk 1]
```diff
@@ -612,11 +612,11 @@ def stft(input: Tensor, n_fft: int, hop_length: Optional[int] = None,
             a real tensor with an extra last dimension for the real and
             imaginary components.
 
-            .. versionchanged:: 1.14.0
+            .. versionchanged:: 2.0.0
```
Comment (lezcano) [inline torch/functional.py:615]: nit.
```suggestion
            .. versionchanged:: 2.0
```

**Review by lezcano — APPROVED** [review]

**Comment by lezcano** [comment]
@pytorch merge -f "rocm failures are unrelated"

**Pull request closed without merging**